ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-eus-4.12.yaml

```diff
@@ -60,15 +60,35 @@ resources:
       cpu: 100m
       memory: 200Mi
 tests:
-- as: azure-ipi-workers-rhel8-f28
-  cron: 6 9 3 * *
+- as: aws-ipi-private-shared-vpc-phz-sts-f14
```

Comment:
As we [discussed](https://redhat-internal.slack.com/archives/CF8SMALS1/p1723516885639519), all eus jobs will be renamed to cpou, and considering there will not be even or odd versions difference for cpou upgrade(only 2-hops and 3-hops difference), let's try to keep the matrix to be consistent with cpou 2-hops jobs. In this way, it could 

1. reduce our maintenance workload across different versions
2. help us compare the test result on even vs odd update path, z-stream vs pre-release update path

so the rule here is:

- cover amd64/arm64/multi payload
- one profile on each platform

Reply:
@jiajliu  Sorry, I couldn't understand you very clearly. do you mean we should keep the google sheet consistent with cpou n-hops? If yes, my answer is: I will DELETE `EUS Upgrade` tag in the sheet as @shellyyang1989 suggested in slack thread and we will track all test matrix in individual JIRA tasks.

Reply:
about `all eus jobs will be renamed to cpou`, do you mean update the config file name not job name?

Reply:
I mean we should use the same profiles for all cpou/eus upgrade. For example: there are following jobs in 4.15-4.17 copou jobs(refer to the matrix in the [jira](https://issues.redhat.com/browse/OTA-1196?focusedId=25259708&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25259708)):
```
A, B in amd64
C, D in arm64
E, F in multi
```
Then let's use the same matrix in 4.16-4.18/4.17-4.19 ... Considering 4.16-4.18 will be copied from 4.14-4.16/4.12-4.14, so let's change it in this pr directly for 4.12-4.14. 
There is only one exception for rhel. Due to I heard of rhel will be deprecated, so I did not add it in 4.17 cpou jobs currently, but for 4.12-4.14 we still need cover it, so we could change it to 
```
A+rhel B in amd64
C D in arm64
E F in multi
```

Reply:
ok, I see. I choose these jobs is because they are the best choice from existing e2e jobs for ocp 4.12. I checked your test matrix, most of the profiles are not exists in 4.12 e2e job, and we are follow the rule that copy e2e job for related upgrade job, so your test matrix are not proper for 4.12 -> 4.14 upgrade. cc @shellyyang1989 and @jianlinliu to confirm

Reply:
> the rule that copy e2e job for related upgrade job

The rule generally works for copy-paste mode in y-stream/z-stream upgrade ci jobs due to the purpose is to ensure full-set coverage on all e2e jobs. But cpou/eus jobs is a little different which is not needed to do full-set coverage,  we only choose a subset of all supported installation profiles.

> most of the profiles are not exists in 4.12 e2e job

- vsphere-ipi-sdn	amd64  //in 4.12
- ibmcloud-ipi-ovn-ipsec   amd64 //not in 4.12 
- nutanix-ipi-proxy-fips-mini-perm   amd64  //not in 4.12
- aws-ipi-ovn-ipsec-arm-mixarch    multi  //in 4.12
- azure-ipi-arm-mixarch multi  //in 4.12
- gcp-ipi-proxy-private	arm64 //in 4.12`
- baremetalds-ipi-ovn-ipv4	arm64   //in 4.12

cc @jianlinliu to help confirm `ibmcloud-ipi-ovn-ipsec` and `nutanix-ipi-proxy-fips-mini-perm` supported or not in 4.12

> test matrix are not proper for 4.12 -> 4.14 upgrade
As above, If we see any profile does not support in 4.12, let's change it to a proper one to replace it with the same platform.

Reply:
multi GAed at ocp 4.13

Reply:
$ grep -rE "aws-ipi-ovn-ipsec-arm-mixarch|azure-ipi-arm-mixarch" *4.12__multi*
openshift-openshift-tests-private-release-4.12__multi-nightly.yaml:- as: aws-ipi-ovn-ipsec-arm-mixarch-f28
openshift-openshift-tests-private-release-4.12__multi-nightly.yaml:- as: azure-ipi-arm-mixarch-f28

btw, there is still 4.12 multi payload delivered periodly https://multi.ocp.releases.ci.openshift.org/releasestream/4-stable-multi/release/4.12.61

Reply:
This page describes that multi-arch first supported in ocp 4.13 https://docs.openshift.com/container-platform/4.13/release_notes/ocp-4-13-release-notes.html#ocp-4-13-updating-clusters-migrating-cluster-to-multi-arch,
And in ocp 4.12 it is TP: https://docs.openshift.com/container-platform/4.12/release_notes/ocp-4-12-release-notes.html#ocp-4-12-ocp-with-multi-arch-compute-machines

let us double confirm with multi team @LiangquanLi930

Reply:
cc @aleskandro

Reply:
As discussed in https://redhat-internal.slack.com/archives/CF8SMALS1/p1723712201102589, multi-arch is not necessary on ocp 4.12, so we could choose other profiles on amd or arm.

Reply:
> to help confirm ibmcloud-ipi-ovn-ipsec and nutanix-ipi-proxy-fips-mini-perm supported or not in 4.12

I suggest to keep align with nightly ci profiles on source version to help us easily get the supported profiles. For the specific two profiles, I did not see them in 4.12 source ci, so I guess they are not supported.

Reply:
> multi-arch is not necessary on ocp 4.12, so we could choose other profiles on amd or arm.

Could u replace it with the same platform in arm? I see there is only one profile baremetal in arm which has limited resource.

Reply:
1. about `azure-ipi-arm-mixarch`, because we don't want to test multi payload in 4.12 -> 4.14 upgrade path, so I think any `azure-ipi` can cover this. right now we have `azure-ipi-workers-rhel8-f14`
2. about `aws-ipi-ovn-ipsec-arm-mixarch`, there is no `aws-ipi-ovn-ipsec` in both amd64 and arm64 e2e config file, so we choose other platform to cover `ovn-ipsec`, right now it is `vsphere-ipi-ovn-ipsec-f14`

cc @shellyyang1989 to confirm

Reply:
Replaced `aws-ipi-private-shared-vpc-phz-sts-f14` in amd nightly with `aws-ipi-proxy-sts-f14` in arm stable.

Reply:
> `aws-ipi-proxy-sts-f14` in arm stable

could u add it in arm nightly? for eus test, stable-nightly sounds enough, we don't have plan to add stable-stable eus upgrade currently.

Reply:
related e2e job only exists in stable config file:
```
[jianl@jianl-thinkpadt14gen4 release]$ grep -e "- as: aws-ipi-proxy-sts" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__arm64-nightly.yaml
[jianl@jianl-thinkpadt14gen4 release]$ grep -e "- as: aws-ipi-proxy-sts" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__arm64-stable.yaml
- as: aws-ipi-proxy-sts-f28
[jianl@jianl-thinkpadt14gen4 release]$ 
`

Reply:
and there are no other aws jobs in nightly config file are proper for eus upgrade:
```
[jianl@jianl-thinkpadt14gen4 release]$ grep -e "- as: aws" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-
private-release-4.12__arm64-nightly.yaml
- as: aws-ipi-disc-priv-sno-basecap-none-f28
- as: aws-ipi-disc-priv-sno-basecap-none-f28-destructive
- as: aws-ipi-disc-priv-sts-efs-f28
- as: aws-ipi-disc-priv-sts-efs-f28-destructive
- as: aws-ipi-disc-priv-sts-ep-f28
- as: aws-ipi-disc-priv-sts-ep-f28-destructive
- as: aws-ipi-disc-priv-tp-f28
- as: aws-ipi-disc-priv-tp-f28-destructive
- as: aws-ipi-f28-longduration-apiserver
- as: aws-ipi-f28-longduration-part1-3
- as: aws-ipi-f28-longduration-part2-3
- as: aws-ipi-private-f28-longduration-apiserver
[jianl@jianl-thinkpadt14gen4 release]$ 
```

Reply:
> related e2e job only exists in stable config file

I'm not sure why it existed in stable but not nightly config(guessing maybe merged into other profiles in nightly day by day). The profiles in stable and nightly should be general available. If we could not search an expected one in nightly, then the profile in stable could also be used. They are the same for upgrade jobs, but adding one more stable-stable job sounds over-workload here.

Reply:
The job moved to arm nightly config file although there is no such job in e2e file.